includes/base/widget-base.php

```diff
@@ -1075,6 +1074,39 @@ public function get_responsive_widgets() {
 		return $responsive_widgets_data_manager->get_asset_data_from_config( $config );
 	}
 
+	/**
+	 * Mark widget as deprecated.
+	 *
+	 * Use `get_deprecation_message()` method to print the message control at specific location in register_controls().
+	 *
+	 * @param $version string           The version of Elementor that deprecated the widget.
+	 * @param $message string         A message regarding the deprecation.
+	 * @param $replacement string   The widget that should be used instead.
+	 */
+	protected function add_deprecation_message( $version, $message, $replacement ) {
+		// Expose the config for handling in JS.
+		$this->set_config( 'deprecation', [
+			'version' => $version,
+			'message' => $message,
+			'replacement' => $replacement,
+		] );
```

Comment:
Should we handle js deprecation if the replacement is not active?

Reply:
No, but it shouldn't be a part of this method, fixed.